### PR TITLE
RFR: Add friends of behat components

### PR DESCRIPTION
**Pull request by PierreRebeilleau**
Use `friends-of-behat/symfony-extension` instead of `behat/symfony2extension`

### CODE REVIEW

File: behat.yml.dist [hunk 1]
```diff
@@ -2,40 +2,35 @@ default:
...
+    'FriendsOfBehat\SymfonyExtension':
+      bootstrap: 'tests/Functional/app/bootstrap.php'
       kernel:
-        env: 'test'
-        debug: 'true'
+        class: 'AppKernel'
         path: 'tests/Functional/app/AppKernel.php'
-        bootstrap: 'tests/Functional/app/bootstrap.php'
+        environment: 'test'
+        debug: true
```
Comment (vincentchalamon) [inline behat.yml.dist:17]:
```suggestion
        debug: false
```

File: behat.yml.dist [hunk 2]
```diff
@@ -2,40 +2,35 @@ default:
...
     'Behat\MinkExtension':
       base_url: 'http://example.com/'
       files_path: 'features/files'
       sessions:
         default:
```
Comment (vincentchalamon) [inline behat.yml.dist:22]:
```suggestion
        symfony:
```

File: features/bootstrap/DoctrineContext.php [hunk 3]
```diff
@@ -40,6 +42,9 @@ final class DoctrineContext implements Context
      * Every scenario gets its own context instance.
      * You can also pass arbitrary arguments to the
      * context constructor through behat.yml.
+     * @param ManagerRegistry $doctrine
```
Comment (vincentchalamon) [inline features/bootstrap/DoctrineContext.php:45]: Useless PHPDoc thanks to typehint

File: features/bootstrap/JsonContext.php [hunk 4]
```diff
@@ -39,8 +42,10 @@ public function createDatabase(BeforeScenarioScope $scope): void
 
     /**
      * @Then /^the JSON should be deep equal to:$/
+     * @param PyStringNode $content
```
Comment (vincentchalamon) [inline features/bootstrap/JsonContext.php:45]: Useless PHPDoc thanks to typehint

File: features/bootstrap/JsonContext.php [hunk 5]
```diff
@@ -61,8 +66,9 @@ public function theJsonShouldBeDeepEqualTo(PyStringNode $content)
 
     /**
      * @Then /^the JSON should be a superset of:$/
+     * @param PyStringNode $content
```
Comment (vincentchalamon) [inline features/bootstrap/JsonContext.php:69]: Useless PHPDoc thanks to typehint

File: features/bootstrap/JsonContext.php [hunk 6]
```diff
@@ -91,8 +97,10 @@ private function sortArrays($obj)
 
     /**
      * @Then the JSON should be valid according to the schema file :file
+     * @param string $file
```
Comment (vincentchalamon) [inline features/bootstrap/JsonContext.php:100]: Useless PHPDoc thanks to typehint

File: features/bootstrap/JsonContext.php [hunk 7]
```diff
@@ -104,8 +112,10 @@ public function theJsonShouldBeValidAccordingToTheSchemaFile(string $file)
 
     /**
      * @Then the JSON should be an array with each entry valid according to the schema file :file
+     * @param string $file
```
Comment (vincentchalamon) [inline features/bootstrap/JsonContext.php:115]: Useless PHPDoc thanks to typehint

File: tests/Functional/app/AppKernel.php [hunk 8]
```diff
@@ -31,7 +31,7 @@ public function __construct(string $environment, bool $debug)
     {
         parent::__construct($environment, $debug);
 
-        // patch for behat/symfony2-extension not supporting %env(APP_ENV)%
+        // patch for Behat/symfony2-extension not supporting %env(APP_ENV)%
         $this->environment = $_SERVER['APP_ENV'] ?? $environment;
```
Comment (vincentchalamon) [inline tests/Functional/app/AppKernel.php:35]: Is it still required?

File: tests/Functional/app/config/services_test.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    arguments:
+      $doctrine: '@doctrine'
+      $jwtManager: '@lexik_jwt_authentication.jwt_manager'
+      $iriConverter: '@api_platform.iri_converter'
+
+  Silverback\ApiComponentBundle\Features\Bootstrap\JsonContext:
+    arguments:
+      $httpCallResultPool: Behatch\HttpCall\HttpCallResultPool
+
+  Behatch\HttpCall\HttpCallResultPool:
```
Comment (vincentchalamon) [inline tests/Functional/app/config/services_test.yaml:19]: Why do you need to create a service from Behatch objects?

File: features/bootstrap/RestContext.php [hunk 10]
```diff
@@ -11,14 +11,28 @@
...
 use Behat\Gherkin\Node\PyStringNode;
 use Behatch\Context\RestContext as BaseRestContext;
 
 /**
  * @author Daniel West <daniel@silverback.is>
  */
-class RestContext extends BaseRestContext
+class RestContext implements Context
 {
+    private BaseRestContext $restContext;
```
Comment (vincentchalamon) [inline features/bootstrap/RestContext.php:26]:
```suggestion
    private ?BehatchRestContext $restContext = null;
```

File: tests/Functional/app/AppKernel.php (hunk 8 above)
Comment (vincentchalamon) [inline tests/Functional/app/AppKernel.php:35]: Is it still required?

File: tests/Functional/app/config/services_test.yaml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+services:
+  _defaults:
+    autowire: true
+    autoconfigure: true
+
+  Silverback\ApiComponentBundle\Features\Bootstrap\:
+    resource: '../../../../features/bootstrap/*'
+
+  Silverback\ApiComponentBundle\Features\Bootstrap\DoctrineContext:
+    arguments:
```
Comment (vincentchalamon) [inline tests/Functional/app/config/services_test.yaml:10]: There is no need to inject those arguments, as autowiring injects them automatically. Please set DoctrineContext as following:
```php
public function __construct(ManagerRegistry $doctrine, JWTManagerInterface $jwtManager, IriConverterInterface $iriConverter)
```

File: features/bootstrap/JsonContext.php [hunk 12]
```diff
@@ -11,24 +11,31 @@
...
+    /**
+     * @BeforeScenario
+     */
+    public function gatherContexts(BeforeScenarioScope $scope): void
     {
-        parent::__construct($httpCallResultPool);
+        $this->jsonContext = $scope->getEnvironment()->getContext(BehatchJsonContext::class);
     }
 
+    public array $components = [];
```
[the remark this answers is not in the document]
Comment (PierreRebeilleau) [reply features/bootstrap/JsonContext.php:36]: Inside the methode `iSaveTheResponseIriAs` but it was commented by @silverbackdan 
Should i remove it ?

File: features/bootstrap/JsonContext.php [hunk 13]
```diff
@@ -11,24 +11,31 @@
...
+    private RestContext $restContext;
```
[the remark this answers is not in the document]
Comment (PierreRebeilleau) [reply features/bootstrap/JsonContext.php:37]: Same as https://github.com/silverbackis/ApiComponentBundle/pull/30#discussion_r410214271

File: features/bootstrap/JsonContext.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply features/bootstrap/JsonContext.php:36]: Yes please, to keep the code clean

File: features/bootstrap/JsonContext.php (hunk 13 above)
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply features/bootstrap/JsonContext.php:37]: Same also

File: tests/Functional/app/config/services_test.yaml [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+      $jwtManager: '@lexik_jwt_authentication.jwt_manager'
```
Comment (chalasr) [inline tests/Functional/app/config/services_test.yaml:11]: Using [`JWTTokenManagerInterface`](https://github.com/lexik/LexikJWTAuthenticationBundle/blob/master/Services/JWTTokenManagerInterface.php) for your constructor argument type (and not `JWTManagerInterface` which is deprecated) would make it autowired, which would avoid the need for defining this service explicitly

Comment (PierreRebeilleau) [reply tests/Functional/app/config/services_test.yaml]: Thanks !

**Comment by silverbackdan** [comment]
Thank you all for your work on this PR.

**Pull request merged**